econml/_ortho_learner.py

```diff
@@ -714,6 +787,64 @@ def score(self, Y, T, X=None, W=None, Z=None, sample_weight=None):
     def model_final(self):
         return self._model_final
 
+    @model_final.setter
+    def model_final(self, model_final):
+        self._model_final = clone(model_final, safe=False)
+
+    @property
+    def model_nuisance(self):
+        return self._model_nuisance
+
+    @model_nuisance.setter
+    def model_nuisance(self, model_nuisance):
+        self._model_nuisance = clone(model_nuisance, safe=False)
+        self._cache_invalid_message = "Setting the nuisance model invalidates the cached nuisances"
+
+    @property
+    def n_splits(self):
+        return self._n_splits
+
+    @n_splits.setter
+    def n_splits(self, n_splits):
+        self._n_splits = n_splits
+        self._cache_invalid_message = "Setting the number of splits invalidates the cached nuisances"
+
+    @property
+    def discrete_treatment(self):
+        return self._discrete_treatment
+
+    @discrete_treatment.setter
+    def discrete_treatment(self, discrete_treatment):
```

Comment:
Does this need to trigger a changed in the discrete treatment transformer??

Reply:
As discussed offline, I don't think so because the transformation happens at fit time.

Reply:
I found a related bug, not sure if related to the transformer. So the following works:
```python
est = LinearDML(model_y=RandomForestRegressor(),
                model_t=RandomForestClassifier(min_samples_leaf=10),
                discrete_treatment=False,
                linear_first_stages=False,
                n_splits=6)
est.fit(Y, T, X=X, W=W)
te_pred = est.effect(X_test)
te_pred[0]
```

But the following raises an error, while from the current semantics it's supposed to work fine:
```python
est = LinearDML(model_y=RandomForestRegressor(),
                model_t=RandomForestClassifier(min_samples_leaf=10),
                discrete_treatment=True,
                linear_first_stages=False,
                n_splits=6)
est.fit(Y, T, X=X, W=W)
est.discrete_treatment = False
est.fit(Y, T, X=X, W=W)
te_pred[0]
```

Which means that there is some state of the object that is maintained but should have been reset when resetting the discrete_treatment.

Reply:
Hm.  I wouldn't expect that to work since it uses a classifier for model_t but a continuous treatment.  I think changing discrete_treatment would require that either the model_t is 'auto' (although I need to fix this logic) or else that the model also changes.

Reply:
Thats why I added the first example that when called a priori it works. The classifier can handle the discrete treatment. So there seems a problem with changing diwcrete treatment

Reply:
Turns out this was due to a typo in property name for the DML setter.  It will work after my next push.